### PR TITLE
Preview.vala: rewrite

### PR DESCRIPTION
**Pull request by danrabbit**
* GObject-style
* Use grid instead of a single label for even spacing with arbitrary formats
* Margins in LocaleSetting.vala instead of internal to this widget
* Remove unused LocaleManager
* Don't set sensitive for legibility

Before:

screenshot from 2017-08-09 10 06 04

After:

screenshot from 2017-08-09 10 04 37

### CODE REVIEW

File: src/Widgets/Preview.vala [hunk 1]
```diff
@@ -43,17 +58,15 @@ namespace SwitchboardPlugLocale.Widgets {
...
             Monetary.strfmon (currency, "%5.2n", 1234.56);
 
-            second_line.set_label ("%s   %s   %'.2f ".printf (
-                date.format("%x   %X"),
-                (string) currency,
-                1234.56));
+            date_label.label = date.format ("%x");
+            time_label.label = date.format ("%X");
+            currency_label.label = (string) currency;
+            number_label.label = "%'.2f".printf ((string) currency, 1234.56);
```
Comment (matzipan) [inline src/Widgets/Preview.vala]: Can you clarify why all this complication is needed?

Comment (danrabbit) [inline src/Widgets/Preview.vala]: In master this is put into a single label with hardcoded literal space characters. This gives what you see above where it's left aligned and the width of the box is not really aligned to anything and the spacing is just arbitrary.

In this branch, I'm using a flowbox so we can have justify-alignment no matter what the font size is set to or which format is currently being displayed.

Comment (matzipan) [inline src/Widgets/Preview.vala]: Oh sorry, I am referring strictly to line 69.

**Comment by matzipan** [comment]
screenshot from 2017-08-13 18 40 08

I installed this plugin and for some odd reason (I can't figure it out), if I click on one of the labels, there's a rounded gray box appear around it.

**Comment by danrabbit** [comment]
Fixed. It's because flowbox children are selectable. I switched to a grid with hexpanded children instead.

**Comment by matzipan** [comment]
Any word on the other comment on line 69?

File: src/Widgets/Preview.vala (hunk 1 above)
Comment (danrabbit) [inline src/Widgets/Preview.vala]: :man_shrugging: it's unchanged from master. Just on one line instead of two

Comment (matzipan) [inline src/Widgets/Preview.vala]: You don't need the `(string) currency` part, AFAICT. It could be replaced with `"%'.2f".printf (1234.56);`

Comment (danrabbit) [inline src/Widgets/Preview.vala]: Yep you're right. I have no idea why that was in there.  Fixed

**Pull request merged**